src/coreclr/vm/jitinterface.h

```diff
@@ -565,20 +565,220 @@ class CEEInfo : public ICorJitInfo
+        CONTRACTL {
+            NOTHROW;
+            GC_NOTRIGGER;
+        } CONTRACTL_END;
+
+        if (m_CodeHeaderRW != m_CodeHeader)
+            freeArrayInternal(m_CodeHeaderRW);
+
+        m_CodeHeader = NULL;
+        m_CodeHeaderRW = NULL;
+
+        m_codeWriteBufferSize = 0;
+        m_pRealCodeHeader = NULL;
+        m_pCodeHeap = NULL;
+
+        if (m_pOffsetMapping != NULL)
+            freeArrayInternal(m_pOffsetMapping);
+
+        if (m_pNativeVarInfo != NULL)
+            freeArrayInternal(m_pNativeVarInfo);
+
+        m_iOffsetMapping = 0;
+        m_pOffsetMapping = NULL;
+        m_iNativeVarInfo = 0;
+        m_pNativeVarInfo = NULL;
+
+        if (m_inlineTreeNodes != NULL)
+            freeArrayInternal(m_inlineTreeNodes);
+        if (m_richOffsetMappings != NULL)
+            freeArrayInternal(m_richOffsetMappings);
+
+        m_inlineTreeNodes = NULL;
+        m_numInlineTreeNodes = 0;
+        m_richOffsetMappings = NULL;
+        m_numRichOffsetMappings = 0;
+    }
+
+    virtual void BackoutJitData(EECodeGenManager * jitMgr) = 0;
+
+    // ICorDebugInfo stuff.
+    void setBoundaries(CORINFO_METHOD_HANDLE ftn,
+                       ULONG32 cMap, ICorDebugInfo::OffsetMapping *pMap) override final;
+    void setVars(CORINFO_METHOD_HANDLE ftn, ULONG32 cVars,
+                 ICorDebugInfo::NativeVarInfo *vars) override final;
+    void CompressDebugInfo();
+    virtual void SetDebugInfo(PTR_BYTE pDebugInfo) = 0;
+
+    virtual PatchpointInfo* GetPatchpointInfo()
+    {
+        return NULL;
+    }
+
+    virtual BOOL JitAgain()
+    {
+        return FALSE;
+    }
+
+    void reportRichMappings(
+        ICorDebugInfo::InlineTreeNode*    inlineTreeNodes,
+        uint32_t                          numInlineTreeNodes,
+        ICorDebugInfo::RichOffsetMapping* mappings,
+        uint32_t                          numMappings) override final;
+
+    void reportMetadata(const char* key, const void* value, size_t length) override final;
+
+    virtual void WriteCode(EECodeGenManager * jitMgr) = 0;
+
+protected:
+
+    template <typename TCodeHeader>
+    void setEHcountWorker(unsigned cEH);
+
+    template<class TCodeHeader>
+    void SetRealCodeHeader();
+
+    template<class TCodeHeader>
+    void NibbleMapSet();
+
+    virtual void getEHinfo(
+                            CORINFO_METHOD_HANDLE ftn,              /* IN  */
+                            unsigned      EHnumber,                 /* IN */
+                            CORINFO_EH_CLAUSE* clause               /* OUT */
+                          ) override final;
+
+    void setEHinfoWorker(
+                          EE_ILEXCEPTION* pEHInfo,
+                          unsigned      EHnumber,
+                          const CORINFO_EH_CLAUSE* clause
+                        );
+
+    EECodeGenManager*       m_jitManager;   // responsible for allocating memory
+    void*                   m_CodeHeader;   // descriptor for JITTED code - read/execute address
```

Comment:
This is no longer restricted to jitted code, right? It also will be there for interpreted code, just not R2R?

Reply:
Right, this is used for the interpreted code too.